src/intertypes/jsonschema.jl

```diff
@@ -0,0 +1,177 @@
+    )
+    I64 => Object(
+      :type => "string",
+      Symbol("\$comment") => "I64"
+    )
+    U64 => Object(
+      :type => "string",
+      Symbol("\$comment") => "U64"
+    )
+    F64 => Object(
+      :type => "number",
+      Symbol("\$comment") => "F64"
+    )
+    Boolean => Object(
+      :type => "boolean",
+      Symbol("\$comment") => "Boolean"
+    )
+    Str => Object(
+      :type => "string",
+      Symbol("\$comment") => "Str"
+    )
+    Sym => Object(
+      :type => "string",
+      Symbol("\$comment") => "Sym"
+    )
+    Binary => Object(
+      :type => "string",
+      :contentEncoding => "base64",
+      Symbol("\$comment") => "Binary"
+    )
+    OptionalType(elemtype) => begin
+      schema = tojsonschema(elemtype)
+      schema[:type] = [schema[:type], "null"]
+      schema
+    end
+    ObjectType(elemtype) => Object(
+      :type => "object",
+      :additionalProperties => tojsonschema(elemtype)
+    )
+    List(elemtype) => Object(
+      :type => "array",
+      :items => tojsonschema(elemtype)
+    )
+    Map(keytype, valuetype) => Object(
+      :type => "array",
+      :items => Object(
+        :type => "object",
+        :properties => Object(
+          :key => tojsonschema(keytype),
+          :value => tojsonschema(valuetype)
+        )
+      )
+    )
+    Record(fields) => recordtype(fields)
+    Sum(variants) => Object(
+      "oneOf" => Vector{Object}(varianttype.(variants))
+    )
+    Annot(desc, innertype) => begin
+      innerschematype = tojsonschema(innertype)
+      innerschematype["description"] = desc
+      innerschematype
+    end
+    TypeRef(to) => reftype(string(toexpr(to)))
+  end
+end
+
+reftype(name) = Object(
+  Symbol("\$ref") => "#/\$defs/$(name)"
+)
+
+recordtype(fields) = Object(
+  :type => "object",
+  :properties => Object(fieldproperties(fields)...),
+  :required => string.(nameof.(fields))
+)
+
+varianttype(variant) = Object(
+  :type => "object",
+  :properties => Object(
+    :tag => Object(
+      :const => string(variant.tag)
+    ),
+    fieldproperties(variant.fields)...
+  ),
+  :required => string.(nameof.(variant.fields))
+)
+
+function acsettype(spec)
+  tablespecs = map(objects(spec.schema)) do ob
+    idfield = Field{InterType}(:_id, U32)
+    homfields = map(homs(spec.schema; from=ob, just_names=true)) do f
+      Field{InterType}(f, U32)
+    end
+    attrfields = map(attrs(spec.schema; from=ob)) do (f, _, t)
+      Field{InterType}(f, spec.schema.typing[t])
+    end
+    Field{InterType}(ob, List(Record([idfield; homfields; attrfields])))
+  end
+  Object(
+    :type => "object",
+    :properties => recordtype(tablespecs)
+  )
+end
+
+"""
+    JSONTarget  
+
+Specifies a serialization target of JSON Schema when
+generating a module.
+
+TODO: This should really be called something like JSONSchemaTarget.
+"""
+struct JSONTarget <: SerializationTarget end
+
+# TODO: Should this be ::JSONTarget instead of ::Type{JSONTarget} so
```

Comment:
IMO I think we should probably keep it as is since it somewhat matches the other parts of source:
- https://github.com/AlgebraicJulia/ACSets.jl/blob/f52002961c9c3ba42c40bfca0b0feaf2f6f39ee2/src/ACSetInterface.jl#L63
- https://github.com/AlgebraicJulia/ACSets.jl/blob/f52002961c9c3ba42c40bfca0b0feaf2f6f39ee2/src/Defaults.jl#L8-L18
- https://github.com/AlgebraicJulia/ACSets.jl/blob/f52002961c9c3ba42c40bfca0b0feaf2f6f39ee2/src/ColumnImplementations.jl#L20-L23 (extending a base function)

Reply:
A counterargument though is that the platform-specific options can be contained in the JSONTarget struct instead of extra keywords.